x/gov/codec.proto

```diff
@@ -154,8 +179,26 @@ message CreateTextProposalMsg {
   bytes author = 8 [(gogoproto.casttype) = "github.com/iov-one/weave.Address"];
 }
 
-// CreateTextProposalMsg deletes a governance proposal.
-message DeleteTextProposalMsg {
+// CreateElectorateUpdateProposalMsg creates a new governance proposal to update an electorate.
+message CreateElectorateUpdateProposalMsg {
```

Comment:
(This comment is fine to ignore here, but important when we add more types)
Odd to copy all this, but there is no way in protobuf to extend a message.

Options seems to be:
1. Embed common types as a sub-message
```proto
message CreateElectorateUpdateProposalMsg {
  CommonProposalMsg common = 1;
  repeated Elector diff_electors = 2 [(gogoproto.nullable) = false];
```

2. Embed the varying part as an Any or bytes field (to be parsed later)
```proto
message CreateProposalMsg {
 // Human readable title. Must match `^[a-zA-Z0-9 _.-]{4,128}$`
  string title = 1;
  // Human readable description with 3 to 5000 chars.
  string description = 2;
  // Reference to the electorate to define the group of possible voters.
  bytes electorate_id = 4 [(gogoproto.customname) = "ElectorateID"];
  // Unix timestamp when the proposal starts. Must be in the future.
  int64 start_time = 5 [(gogoproto.casttype) = "github.com/iov-one/weave.UnixTime"];
  // Author is an optional field to set the address of the author with a proposal. The author must sign the message.
  // When not set it will default to the main signer.
  bytes author = 8 [(gogoproto.casttype) = "github.com/iov-one/weave.Address"];

  // kind lets us somehow know how to parse the next field 
  string kind = 10;
  // note that an embedded message is compatible with bytes, just won't be parsed yet (like json.RawMessage type)
  // https://developers.google.com/protocol-buffers/docs/proto3#updating
  bytes data = 11;
}
```

Reply:
There is a fair amount of duplicate fields (and code in the create handler) that require some further refactoring. Both are good and valid solutions to problem. Thanks for bringing them up! The second solution feels like an extra step for both sides but I don't have an opinion, yet. 🤔

Reply:
Both are good ideas and valid options. Thanks for bringing them up! I wonder if example 2 is a bit too complex for the client side but should work. I don't have a preference, yet. 🤔

Reply:
I write the client side bindings.
If it is possible with protobuf, I can do it.

But yeah, the simplest way that works is best... too much complexity has cost